### PR TITLE
Make cyclic import checker respect local disable clauses

### LINKED ISSUE #59: Impossible to disable R0401 with comments.
Originally reported by: **Anonymous**

---

In two files with a circular import, putting R0401 anywhere in the file - at the top, after the line of code in question, etc - would not disable the output. It had to be added to the pylintrc global disable.

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/61

### PR DESCRIPTION
**Pull request by rogalski**
Fixes #59

### CODE REVIEW

**Comment by rogalski** [comment]
Failures are unrelated to changes - looks like there's a regression in astroid.

File: pylint/checkers/imports.py [hunk 1]
```diff
@@ -347,15 +348,23 @@ def open(self):
         self.linter.add_stats(cycles=[])
         self.stats = self.linter.stats
         self.import_graph = collections.defaultdict(set)
+        self._excluded_edges = collections.defaultdict(set)
         self._ignored_modules = get_global_option(
             self, 'ignored-modules', default=[])
 
+    def _import_graph_without_ignored_edges(self):
+        from copy import deepcopy
```
Comment (PCManticore) [inline pylint/checkers/imports.py]: We can have the import at the top of the file.

Comment (rogalski) [inline pylint/checkers/imports.py]: Sure

File: pylint/checkers/imports.py [hunk 2]
```diff
@@ -347,15 +348,23 @@ def open(self):
...
+        filtered_graph = deepcopy(self.import_graph)
+        for node in filtered_graph:
+            filtered_graph[node] -= self._excluded_edges[node]
```
Comment (rogalski) [inline pylint/checkers/imports.py]: Sure

**Pull request merged**